src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/UserDefinedImplicitConversions.cs

```diff
@@ -278,7 +278,8 @@ private static void ComputeUserDefinedImplicitConversionTypeSet(TypeSymbol s, Ty
 
                         u.Add(UserDefinedConversionAnalysis.Normal(op, fromConversion, toConversion, convertsFrom, convertsTo));
                     }
-                    else if ((object)source != null && (object)target != null && source.IsNullableType() && convertsFrom.IsNonNullableValueType() && target.CanBeAssignedNull())
+                    else if ((object)source != null && source.IsNullableType() && convertsFrom.IsNonNullableValueType() &&
+                        ((object)target != null ? target.CanBeAssignedNull() : convertsTo.IsReferenceType))
```

Comment:
Item (1) is already addressed by the `<remarks>` paragraph on the enclosing method, which says

```cs
        /// <remarks>
        /// Currently allowAnyTarget flag is only set to true by AnalyzeImplicitUserDefinedConversionForSwitchGoverningType,
        /// where we must consider user defined implicit conversions from the type of the switch expression to
        /// any of the possible switch governing types.
        /// </remarks>
```

Item (2) is a bug (that I'm about to fix), as can be seen by a new test I'm about to add.